agents/opencode_chat.py

```
@@ -0,0 +1,261 @@
+        if not self.password:
+            return {}
+        creds = b64encode(f"opencode:{self.password}".encode()).decode()
+        return {"Authorization": f"Basic {creds}"}
+
+    def _health_check(self) -> bool:
+        now = time.time()
+        if now - self._last_health_check < self._health_check_interval:
+            return True
+        try:
+            url = f"{self.base_url}/global/health"
+            req = urllib.request.Request(url, headers=self._get_auth_header())
```

Ruff (0.15.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)